### ISSUE TITLE
Invite-only system no longer working

### ISSUE THREAD

**Issue by aendra-rininsland** [maccman/monocle#26]
Howdy!

It seems the invite system was partially removed in commit cdf602df869fd1073502466644079564b759093f. Part of what made me interested in using this project in the first place was the invite system, and it's kinda confusing that there are still relics from that (Limited invites, columns in the User model).

Suggested resolution:
- Re-add the invite system in a more modular fashion, such that it can be turned off and on via config option.
- Failing that, remove the constrained invitation system (I.e., give users unlimited invites).
- Create a branch for using the invitation system, such as is suggested used to be the case in [Dan's amazing Heroku blogpost](http://dan-jenkins.co.uk/monocle-on-heroku/).

**Comment by maccman** [comment]
Yup, I half/removed the invite system (not very cleanly). Apologies.
Definitely happy with a config option for invites.

On Tue, Feb 18, 2014 at 11:32 AM, Ændrew Rininsland <
notifications@github.com> wrote:

> Howdy!
> 
> It seems the invite system was partially removed in commit cdf602dhttps://github.com/maccman/monocle/commit/cdf602df869fd1073502466644079564b759093f.
> Part of what made me interested in using this project in the first place
> was the invite system, and it's kinda confusing that there are still relics
> from that (Limited invites, columns in the User model).
> 
> Suggested resolution:
> - Re-add the invite system in a more modular fashion, such that it can
>   be turned off and on via config option.
> - Failing that, remove the constrained invitation system (I.e., give
>   users unlimited invites).
> 
> ## 
> 
> Reply to this email directly or view it on GitHubhttps://github.com/maccman/monocle/issues/26
> .

## 

Alex MacCaw

+1.415.624.7665
@maccaw

http://alexmaccaw.com

**Comment by aendra-rininsland** [comment]
@maccman No worries! Found the commit where it was done, doesn't seem like it'd be too much work to re-add it with config options. Thanks for getting back to me!

My inexperience with Ruby may be a bit apparent with this, but any particular place where you're doing global config already? Or is it all just env vars? If just env vars, I'll continue in that vein.

**Comment by maccaw** [comment]
ENV var read from app.rb

On Tue, Feb 18, 2014 at 2:20 PM, Ændrew Rininsland <notifications@github.com

> wrote:
> 
> @maccman https://github.com/maccman No worries! Found the commit where
> it was done, doesn't seem like it'd be too much work to re-add it with
> config options. Thanks for getting back to me!
> 
> My inexperience with Ruby may be a bit apparent with this, but any
> particular place where you're doing global config already? Or is it all
> just env vars? If just env vars, I'll continue in that vein.
> 
> ## 
> 
> Reply to this email directly or view it on GitHubhttps://github.com/maccman/monocle/issues/26#issuecomment-35387997
> .

## 

Alex MacCaw

+1.415.624.7665
@maccaw

http://alexmaccaw.com

**Comment by aendra-rininsland** [comment]
@maccaw Cool, sounds good!

I've done a bunch of git stuff so that aendrew/newsdevlist#master is a straight copy of maccman/monocle's master branch and [aendrew/newsdevlist#readding_invites](https://github.com/aendrew/newsdevlist/tree/readding_invites) has my changes re-adding user activation against commit 56f4639ef652dd5bf168cf5d9d5c7bbc7ab2d79a. 

Once I add the system configurable via env vars I'll do a pull request.
